workflows/bacterial_base.nf

```diff
@@ -22,37 +25,59 @@ workflow CALL_BACTERIAL_BASE {
         coreLociBed
         referenceGenome
         referenceGenomeDir
-        ch_meta_iontorrent
-        ch_meta_illumina
-        ch_meta_nanopore
+        inputSamples
     
     main:
         ch_versions = Channel.empty()
 
-        // reads trim and clean
-        assembly_trim_clean(ch_meta_iontorrent).set{ ch_clean_meta }
-        ch_meta_illumina.mix(ch_clean_meta, ch_meta_nanopore).set{ ch_input_meta }
-        ch_input_meta.map { sampleID, reads, platform -> [ sampleID, reads ] }.set{ ch_reads }
+        // Create channel for sample metadata
+        Channel.fromPath(inputSamples)
+            .splitCsv(header:true)
+            .tap{ ch_raw_input }
+            .map{ row -> get_meta(row) }
+            .set{ ch_meta }
+
+        // Create channel for reads
+        ch_raw_input
+            .map{ row -> get_reads(row) }
+            .set{ ch_reads }
+
+        // downsample reads
+        seqtk_sample( ch_reads.map(row -> [row[0], row[1], params.targetSampleSize]) ).reads
```

Comment:
Are we not separating `params.targetSampleSize` into it's own channel? We agreed that `params.targetSampleSize` is not misleading right?

Reply:
See other comment.